starlette/requests.py

```diff
@@ -220,10 +220,12 @@ async def stream(self) -> typing.AsyncGenerator[bytes, None]:
         while True:
             message = await self._receive()
             if message["type"] == "http.request":
+                if not message.get("more_body", False):
+                    self._stream_consumed = True
```

Comment:
This gets used by `_CachedRequest`, adding it here avoids having to overwrite all of async def stream.